usage.c

```diff
@@ -15,7 +15,12 @@ void vreportf(const char *prefix, const char *err, va_list params)
 
 	fflush(fh);
 	if (!tweaked_error_buffering) {
+#if defined(_MSC_VER)
+		/* UCRT doesn't like zero buffer size */
+		setvbuf(fh, NULL, _IOLBF, BUFSIZ);
+#else
 		setvbuf(fh, NULL, _IOLBF, 0);
+#endif
```

Comment:
The commit that you suggested to revert intended to switch to line buffered mode, nothing more, nothing less. [Linux' man page on `setvbuf`](http://linux.die.net/man/3/setvbuf) states:

> If the argument `buf` is `NULL`, only the mode is affected;

In other words, the size *is* ignored if `buf` is set to `NULL`. It gets worse: the man page then goes on to explain the `setlinebuf()` variant (which conforms only to 4.2BSD, hence we cannot use it):

>  The `setlinebuf()` function is exactly equivalent to the call:
> ```c
>     setvbuf(stream, NULL, _IOLBF, 0); 
> ```

Please note that this is precisely what Git's source code reads at the moment.

As you can see, the documentation you are most likely to find on the 'net disagrees with this statement:

> But the documentation does not say that the function supports the zero size, so _IOLBF should be always with non-zero size as you suggesting